File: C++/Flatten BST to sorted list.cpp
class Solution
{
public:
Node *head=NULL;
Node *tail=head;
void solve(Node *root){
    if(root==NULL)
    return;
    solve(root->left);
     if(tail==NULL){
       head=root;
    }
     if(tail!=NULL){
         tail->right=root;
         tail->left=NULL;
     }
   tail=root;
   solve(root->right);
   tail->right=NULL;
   tail->left=NULL;
}
    Node *flattenBST(Node *root)
    {
        solve(root);
        return head;
    }
};

File: JAVA/Flatten BST to sorted list.java
class Solution {
    public Node flattenBST(Node root) {
        Node dummy = new Node(-1);
        Node current = dummy;
        Stack<Node> stack = new Stack<>();
        Node node = root;
        while (node != null || !stack.isEmpty()) {
            while (node != null) {
                stack.push(node);
                node = node.left;
            }
            node = stack.pop();
            current.right = node;
            current = current.right;
            node.left = null; // Set left child to null
            node = node.right;
        }
        return dummy.right;
    }
}
